doc/xml/parsing.md

```diff
@@ -0,0 +1,504 @@
+# Parsing
+
+This page shows how \Nokogiri parses an XML string into \Nokogiri objects.
+The string has text consisting of character data and markup.
+For a \Nokogiri parsing method, the string is passed
+either as a [String](https://docs.ruby-lang.org/en/master/String.html) object
+or as an [IO](https://docs.ruby-lang.org/en/master/IO.html) object
+from which the string is to be read.
+
+Most of the sections below link to a relevant section in the W3C document
+[Extensible Markup Language (XML) 1.0 (Fifth Edition)](https://www.w3.org/TR/REC-xml/).
+
+
+On this page, each example uses either:
+
+- Method Nokogiri::XML::parse (shorthand for Nokogiri::XML::Document.parse)
+  to parse a string into a tree of \Nokogiri objects.
+  The topmost object is a Nokogiri::XML::Document object,
+  which we will usually refer to as a document;
+  the document may have other objects as children.
```

Comment:
Can you please clarify this section regarding the number of children? The most notable difference between a Document and a DocumentFragment is that the Document only has one child at the root level, the root element, while a DocumentFragment may have more than one child at the root level.

Reply:
Clarifying (I hope) remarks added.